crates/estree/src/util.rs

```diff
@@ -0,0 +1,97 @@
+use serde_json::Value;
+use serde::ser::*;
+
+pub enum Container {
+    NothingMeaningful,
+    FunctionDeclaration,
+    FunctionExpression,
+}
+
+/// A container used to represent some piece of data being
+/// serialized. Future versions may add options to the
+/// serialization format.
+pub struct Serialization<'a, T> where T: 'a {
```

Comment:
No, I didn't mean for easter, but just return a wrapper that consumer can further call `.serialize` on.

> There are at least two widely-used versions of ESTree available: Esprima's and Babel's.

Babel has own AST format which they forked from ESTree, but it's not ESTree anymore which is clearly defined in the [spec](https://github.com/estree/estree).

However, I don't see it as a big problem to just return result of `json!` macro (essentially a `Map`) and then consumer can just provide own `Serializer` instead that would transform nodes before writing them out as strings, or accept serialization options or anything else.